sonar-cxx-plugin/src/main/java/org/sonar/plugins/cxx/squid/CxxSquidSensor.java

```diff
@@ -175,99 +153,111 @@ private CxxConfiguration createConfiguration(FileSystem fs, Settings settings) {
-    context.saveMeasure(inputFile, CoreMetrics.COMPLEXITY_IN_FUNCTIONS, complexityInFunctions);
+    context.<Integer>newMeasure().forMetric(CoreMetrics.COMPLEXITY_IN_CLASSES).on(inputFile).withValue(complexityInClasses).save();
+    context.<Integer>newMeasure().forMetric(CoreMetrics.COMPLEXITY_IN_FUNCTIONS).on(inputFile).withValue(complexityInFunctions).save();
   }
 
-  private void saveFilesComplexityDistribution(InputFile inputFile, SourceFile squidFile) {
-    RangeDistributionBuilder complexityDistribution = new RangeDistributionBuilder(CoreMetrics.FILE_COMPLEXITY_DISTRIBUTION, FILES_DISTRIB_BOTTOM_LIMITS); //@todo deprecated RangeDistributionBuilder
+  private void saveFilesComplexityDistribution(InputFile inputFile, SourceFile squidFile, SensorContext context) {    
+    RangeDistributionBuilder fileComplexityDistribution = new RangeDistributionBuilder(LIMITS_COMPLEXITY_FILES);
     double complexity = squidFile.getDouble(CxxMetric.COMPLEXITY);
-    complexityDistribution.add(complexity);
-    context.saveMeasure(inputFile, complexityDistribution.build().setPersistenceMode(PersistenceMode.MEMORY));
+    fileComplexityDistribution.add(complexity);    
+    context.<String>newMeasure().forMetric(CoreMetrics.FILE_COMPLEXITY_DISTRIBUTION).on(inputFile).withValue(fileComplexityDistribution.build()).save();
   }
 
-  private int saveViolations(InputFile inputFile, SourceFile squidFile) {
+  private int saveViolations(InputFile inputFile, SourceFile squidFile, SensorContext sensorContext) {
     Collection<CheckMessage> messages = squidFile.getCheckMessages();
     int violationsCount = 0;
     if (messages != null) {
-      Issuable issuable = resourcePerspectives.as(Issuable.class, inputFile);
-      if (issuable != null) {
-        for (CheckMessage message : messages) {
-          Issue issue = issuable.newIssueBuilder()
-            .ruleKey(checks.ruleKey((SquidAstVisitor<Grammar>) message.getCheck()))
-            .line(message.getLine()) //@todo deprecated line
-            .message(message.getText(Locale.ENGLISH)) //@todo deprecated message
-            .build();
-          if (issuable.addIssue(issue)) {
-            violationsCount++;
+      for (CheckMessage message : messages) {
+        try
+        {
+          int line = 1;
+          if (message.getLine() != null && message.getLine() > 0) {
+           line = message.getLine();
           }
+          
+          NewIssue newIssue = sensorContext.newIssue().forRule(RuleKey.of(CheckList.REPOSITORY_KEY, checks.ruleKey((SquidAstVisitor<Grammar>) message.getCheck()).rule()));
+          NewIssueLocation location = newIssue.newLocation()
+            .on(inputFile)
+            .at(inputFile.selectLine(line))
+            .message(message.getText(Locale.ENGLISH));
+
+          newIssue.at(location);
+          newIssue.save();
+        
+        } catch (Exception ex) {
+          CxxUtils.LOG.info("Line '{}'", message.getLine());
```

Comment:
done, remove try catch block (left overs of a crash investigation during implementation)